python/cudf_polars/cudf_polars/experimental/io.py

```diff
@@ -263,7 +263,8 @@ def _sample_pq_statistics(ir: Scan) -> dict[str, np.floating[T]]:
     import numpy as np
 
     # Use average total_uncompressed_size of three files
-    n_sample = min(3, len(ir.paths))
+    n_sample = min(5, len(ir.paths))
+    random.seed(42)  # Deterministic partitioning
```

Comment:
@TomAugspurger @quasiben - I noticed that we can get different partitioning when we sample different files here. May be best to use a deterministic seed for now. Otherwise, it can be difficult to compare two different runs.

Reply:
Agreed we want determinism. We should be careful not to affect the global state of the random module though. Looks like `random.Random()` can maybe be used.

And I guess the intention of not using the first 5 paths is that those might not be representative?

Reply:
>We should be careful not to affect the global state of the random module though.

Yeah, totally agree.

>And I guess the intention of not using the first 5 paths is that those might not be representative?

That was the original reason for the random sampling, but sampling the first `N` may be best for now. It probably makes sense to make the sampling behavior configurable, but I'm hesitant to add this knob until we decide it's necessary.

Reply:
Okay - Just realized we can just choose a deterministic stride to sample 5 files across the path list. That should be sufficient for now.